tests/kafkatest/tests/verifiable_consumer_test.py

```diff
@@ -0,0 +1,83 @@
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from ducktape.utils.util import wait_until
+
+from kafkatest.tests.kafka_test import KafkaTest
+from kafkatest.services.zookeeper import ZookeeperService
+from kafkatest.services.kafka import KafkaService
+from kafkatest.services.verifiable_producer import VerifiableProducer
+from kafkatest.services.verifiable_consumer import VerifiableConsumer
+from kafkatest.services.kafka import TopicPartition
+
+class VerifiableConsumerTest(KafkaTest):
```

Comment:
The verifiable consumer only works with the new consumer, but it provides a little more for test cases to work with. For example, it emits events corresponding to each invocation of the rebalance callback and for the result of every offset commit. It also includes the offsets of consumed messages. 

However, there is clearly some redundancy. I think we could either try to get the verifiable consumer working with the old consumer and change the `ProduceConsumeValidateTest` to use it instead of the console consumer, or we can wait until the old consumer is deprecated and do the same. Unfortunately, some of the test cases are specific to the new consumer, so the latter might be the best choice. We could also extend the console consumer to provide some of the same functionality, but it's useful having a separate component dedicated to testing since we can extend it without complicating a user-facing tool.

The only reason I didn't add a field for the consumer and the producer is that there are often test cases which parameterize one of its settings (e.g. enabling autocommit). Perhaps I could have gotten around this by modifying the fields of the consumer instance, but I felt that left the configuration of the consumer unclear on test instantiation. Perhaps ducktape already provides a way to handle this?